pywren/future.py

```diff
@@ -94,6 +102,10 @@ def result(self, timeout=None, check_only=False, throw_except=True, storage_hand
         if self._state == JobState.new:
             raise ValueError("job not yet invoked")
 
+        if check_only:
```

Comment:
why don't we write a new `check` function that we can call from within this function ? That way we can avoid that `FIXME` below ?

Reply:
Yes that's the right thing to do but the three of us all have deadlines; writing `check()` creates _another_ "temporarily create the storage handler` point in the code and I don't want to think about that right now.  

basically our semantics are crap and we need a refactoring; right now I just want `done()` to work well enough that I can have a fast status check. 

Basically this is meant as a patch not a refactor :) 

I agree though this all stinks, it stinks to high heaven.